enterprise/frontend/src/metabase-enterprise/metabot/components/Metabot.tsx

```diff
@@ -9,13 +9,15 @@ import { useMetabotAgent } from "../hooks";
 
 import { MetabotChat } from "./MetabotChat";
 
-export const Metabot = () => {
+export interface MetabotProps {
+  hide?: boolean;
```

Comment:
Is this prop only used in tests?

Reply:
nvm, It's how we hide for admin. But it feels kinda weird to have 2 ways to hide it, no? One as a prop, one as state?

Reply:
I suppose it's slightly weird, but I don't have access to react router hooks, so some way or another I would have to pass a prop (routerProps, location.pathname, etc.) which I'll then use to hide Metabot with. This is the route I initially took, but when I considered if we might want to reuse Metabot as an SDK component, the more generic `hide` felt like a more flexible choice to me. I could absolutely be swayed though.

Reply:
couldn't you just conditionally render it like

```ts
{!isAdminApp && <Metabot />}
```

and then get rid of all that useEffect nonsense?

Reply:
I could, but I'm wanting to keep the `metabot.visible` value in sync. This is nice because I can just rely on this value when doing things like deciding if the AppBar should have a bottom border w/o having to do some kind of `isMetabotVisible(routerProps, metabotState)` in places like this.

Reply:
I wouldn't block on this issue, but syncing state with useEffect is always hard to reason about.

Reply:
Agreed.